pkg/service/oidc_provider_test.go

```diff
@@ -0,0 +1,186 @@
+	if provider.GetSpec().GetProviderName() != "provider-"+pruuid {
+		t.Error("invalid provider name returned")
+	}
+}
+
+func TestOidcCreateProvider(t *testing.T) {
+	db, mock := getDB(t)
+	defer db.Close()
+
+	OP := NewOIDCProviderService(db, "", getLogger())
+
+	uuuid := uuid.New().String()
+	pruuid := uuid.New().String()
+	puuid, ouuid := addParterOrgFetchExpectation(mock)
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id" FROM "authsrv_oidc_provider" AS "oidcprovider" WHERE .organization_id = '` + ouuid + `'. AND .partner_id = '` + puuid + `'. AND .name = 'user-` + uuuid + `'.`).
+		WillReturnError(fmt.Errorf("no data available"))
+
+	scope := []string{"system"}
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id"`).
+		WillReturnError(fmt.Errorf("no data available here"))
+
+	mock.ExpectQuery(`INSERT INTO "authsrv_oidc_provider"`).
+		WithArgs().WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(puuid))
+
+	provider := &systemv3.OIDCProvider{
+		Metadata: &v3.Metadata{Partner: "partner-" + puuid, Organization: "org-" + ouuid, Name: "user-" + uuuid},
+		Spec:     &systemv3.OIDCProviderSpec{Scopes: scope, IssuerUrl: "https://", ProviderName: "provider-" + pruuid},
+	}
+
+	provider, err := OP.Create(context.Background(), provider)
+	if err != nil {
+		t.Fatal("err:", err)
+	}
+	performOidcProviderBasicChecks(t, provider, uuuid, pruuid)
+}
+
+func TestOidcProviderGetById(t *testing.T) {
+	db, mock := getDB(t)
+	defer db.Close()
+
+	OP := NewOIDCProviderService(db, "", getLogger())
+
+	uuuid := uuid.New().String()
+	pruuid := uuid.New().String()
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id", "oidcprovider"."name"`).
+		WithArgs().WillReturnRows(sqlmock.NewRows([]string{"id", "name", "provider_name"}).AddRow(uuuid, "user-"+uuuid, "provider-"+pruuid))
+
+	provider := &systemv3.OIDCProvider{
+		Metadata: &v3.Metadata{Id: uuuid, Name: "user-" + uuuid},
+	}
+
+	provider, err := OP.GetByID(context.Background(), provider)
+	if err != nil {
+		t.Fatal("could not get provider:", err)
+	}
+
+	t.Log("id: ", provider.GetSpec().GetProviderName())
+
+	performOidcProviderBasicChecks(t, provider, uuuid, pruuid)
+}
+
+func TestOidcProviderGetByName(t *testing.T) {
+	db, mock := getDB(t)
+	defer db.Close()
+
+	OP := NewOIDCProviderService(db, "", getLogger())
+
+	pruuid := uuid.New().String()
+	uuuid := uuid.New().String()
+	uuuuid := uuid.New().String()
+
+	mock.ExpectQuery(`SELECT "oidcprovider"."id", "oidcprovider"."name"`).
+		WithArgs().WillReturnRows(sqlmock.NewRows([]string{"id"}).AddRow(uuuid))
+
+	provider := &systemv3.OIDCProvider{
+		Metadata: &v3.Metadata{Id: uuuid, Name: "user-" + uuuuid},
+		Spec:     &systemv3.OIDCProviderSpec{ProviderName: "provider-" + pruuid},
+	}
+
+	_, err := OP.GetByName(context.Background(), provider)
+	if err != nil {
+		t.Fatal("could not get partner:", err)
+	}
+	performOidcProviderBasicChecks(t, provider, uuuuid, pruuid)
+}
+
+func TestOidcProviderUpdate(t *testing.T) {
```

Comment:
Can you add one more test case with invalid issuer / mapper url to cover validateURL function as well.

Reply:
Started with in `TestOidcProviderUpdateInvalidUrl`